infra/feast-operator/internal/controller/featurestore_controller_volume_volumemount_test.go

```diff
@@ -0,0 +1,215 @@
+			err = k8sClient.Get(ctx, typeNamespacedName, resource)
+			Expect(err).NotTo(HaveOccurred())
+
+			feast := services.FeastServices{
+				Handler: handler.FeastHandler{
+					Client:       controllerReconciler.Client,
+					Context:      ctx,
+					Scheme:       controllerReconciler.Scheme,
+					FeatureStore: resource,
+				},
+			}
+
+			deploy := &appsv1.Deployment{}
+			objMeta := feast.GetObjectMeta()
+			err = k8sClient.Get(ctx, types.NamespacedName{
+				Name:      objMeta.Name,
+				Namespace: objMeta.Namespace,
+			}, deploy)
+
+			Expect(err).NotTo(HaveOccurred())
+
+			// Extract the PodSpec from DeploymentSpec
+			podSpec := deploy.Spec.Template.Spec
+
+			// Validate Volumes
+			// Validate Volumes - Check if our test volume exists among multiple
+			Expect(podSpec.Volumes).To(ContainElement(WithTransform(func(v corev1.Volume) string {
+				return v.Name
+			}, Equal("test-volume"))), "Expected volume 'test-volume' to be present")
+
+			// Ensure 'online' container has the test volume mount
+			var onlineContainer *corev1.Container
+			for i, container := range podSpec.Containers {
+				if container.Name == "online" {
+					onlineContainer = &podSpec.Containers[i]
+					break
+				}
+			}
+			Expect(onlineContainer).ToNot(BeNil(), "Expected to find container 'online'")
+
+			// Validate that 'online' container has the test-volume mount
+			Expect(onlineContainer.VolumeMounts).To(ContainElement(WithTransform(func(vm corev1.VolumeMount) string {
+				return vm.Name
+			}, Equal("test-volume"))), "Expected 'online' container to have volume mount 'test-volume'")
+
+			// Ensure all other containers do NOT have the test volume mount
+			for _, container := range podSpec.Containers {
+				if container.Name != "online" {
+					Expect(container.VolumeMounts).ToNot(ContainElement(WithTransform(func(vm corev1.VolumeMount) string {
+						return vm.Name
+					}, Equal("test-volume"))), "Unexpected volume mount 'test-volume' found in container "+container.Name)
+				}
+			}
+
+		})
+	})
+})
+
+func createFeatureStoreVolumeResource(resourceName string, image string, pullPolicy corev1.PullPolicy) *feastdevv1alpha1.FeatureStore {
+	volume := corev1.Volume{
+		Name: "test-volume",
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		},
+	}
+	volumeMount := corev1.VolumeMount{
+		Name:      "test-volume",
+		MountPath: "/data",
+	}
+
+	return &feastdevv1alpha1.FeatureStore{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      resourceName,
+			Namespace: "default",
+		},
+		Spec: feastdevv1alpha1.FeatureStoreSpec{
+			FeastProject: feastProject,
+			Services: &feastdevv1alpha1.FeatureStoreServices{
+				Volumes: []corev1.Volume{volume},
+				OfflineStore: &feastdevv1alpha1.OfflineStore{
+					ServerConfigs: feastdevv1alpha1.ServerConfigs{
+						ContainerConfigs: feastdevv1alpha1.ContainerConfigs{},
+					},
+				},
+				OnlineStore: &feastdevv1alpha1.OnlineStore{
+					ServerConfigs: feastdevv1alpha1.ServerConfigs{
```

Comment:
several of these will need to change as well
```suggestion
					Server: &feastdevv1alpha1.ServerConfigs{
```